src/imagecaptioning/imagecaptioning.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module image/imagecaptioning/imagecaptioning
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import ImageCaptioningEngine from './imagecaptioningengine';
+import '../../theme/imagecaptioning/theme.scss';
+
+/**
+ * The image captioning plugin.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class ImageCaptioning extends Plugin {
```

Comment:
Besides, since this plugin is empty now, you can merge the engine part into it. Unless... unless we see that adding the placeholder will require some strictly UI-related logic.

OTOH, by splitting the feature beforehand we stick to some pattern, so I'm not sure what would be better.